homeassistant/components/mqtt/camera.py

```diff
@@ -40,20 +45,41 @@
     }
 )
 
-PLATFORM_SCHEMA_MODERN = MQTT_BASE_SCHEMA.extend(
+
+# Using CONF_ENCODING to set b64 encoding for images is deprecated as of Home Assistant 2022.9
+# use CONF_IMAGE_ENCODING instead, support for the work-a-round will be removed with Home Assistant 2022.11
+def repair_legacy_encoding(config: ConfigType) -> ConfigType:
+    """Check incorrect deprecated config of image encoding."""
+    if config[CONF_ENCODING] == "b64":
+        config[CONF_IMAGE_ENCODING] = "b64"
+        config[CONF_ENCODING] = DEFAULT_ENCODING
+        _LOGGER.warning(
+            "Using the `encoding` parameter to set image encoding has been deprecated, use `image_encoding` instead"
+        )
+    return config
+
+
+PLATFORM_SCHEMA_BASE = MQTT_BASE_SCHEMA.extend(
     {
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
         vol.Required(CONF_TOPIC): valid_subscribe_topic,
+        vol.Optional(CONF_IMAGE_ENCODING): cv.matches_regex(r"^(b64)( \d*)?$"),
```

Comment:
Whats up with the regex? Can't this be simplified to just:
```suggestion
        vol.Optional(CONF_IMAGE_ENCODING): "b64",
```